api/v1/routes/product.py

```diff
@@ -175,3 +190,42 @@ def delete_product(
     """
 
     product_service.delete(db=db, id=id, current_user=current_user)
+
+
+@product.post("/", status_code=status.HTTP_201_CREATED, response_model=success_response)
+async def create_product(
+    product_data: ProductCreate,
+    current_user: Annotated[User, Depends(get_current_user)],
```

Comment:
Change this to user_service.get_current_super_admin since u was told that the endpoint is meant for superadmins alone